pkg/build/controller/controller.go

```diff
@@ -125,29 +125,8 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 	// Set the build phase, which will be persisted if no error occurs.
 	build.Status.Phase = buildapi.BuildPhasePending
 
-	// Make a copy to avoid mutating the build from this point on.
```

Comment:
does CreateBuildPod mutate the build at all? Can we add a test that ensures nextBuldPhase doesn't mutate the build in unexpected ways (e.g. only the status phase changes). I'd also like tests for CreateBuildPod for each strategy to make sure they don't change the build (or pass a struct to them, rather than a reference)